### ISSUE TITLE
[SS 3.6.5] Shortcodeparser: ERROR [User Warning]: None of the following templates could be found (no theme in use): ...

### ISSUE THREAD

**Issue by Friksel** [silverstripe/silverstripe-framework#8000]
## Affected Version
3.6.5

## Description
When using the latest version of Silverstripe framework there seem to be something changed in the shortcodeparser causing SSviewer to be unable to know the current theme.
Because of that pages with a custom shortcode using a template cannot be saved anymore and I get the following error when clicking publish or save. I am trying to debug this, but so far without luck. 

```
ERROR [User Warning]: None of the following templates could be found (no theme in use): sc_iframe.ss
IN POST /admin/pages/edit/EditForm/210
Line 918 in {path}\framework\view\SSViewer.php
```
But sc_iframe.ss does exist and was previously (in earlier versions of Silverstripe) perfectly found without a problem. I'm not sure wich Silverstripe version started this problem, but it must be a very recent one, 'cause in all other versions I never seen this problem.

This is the my custom scortcode:
```
ShortcodeParser::get('default')->register('iframe', function($arguments, $parser, $shortcode) {
    $src = (isset($arguments['src']) && $arguments['src']) ? $arguments['src'] : '';

    // some code here to set $orgWidth, $orgHeight, $maxWidth and $mainDivClasses

    $templateData = array(
        'mainDivClasses'=> $mainDivClasses,
        'iFrameURL' => ($src && $src != '') ? $src : false,
        'iFrameWidth' => strval($orgWidth),
        'iFrameHeight' => strval($orgHeight),
        'iFrameMaxWidth' => strval($maxWidth)
    );

    $template = new SSViewer('sc_iframe');
    return $template->process(new ArrayData($templateData));
});
```
The shortcut `[sitetree_link id=1]` still works, but my guess would be that's because it's not using a template?

## Steps to Reproduce
- Using the custom shortcode; Insert this shortcode in a content-field:
[iframe,src="road/to/some/html",orgWidth="1920",orgHeight="1080",scale="1"][/iframe]
- Save/publish page
--> the page isn't saved because of the error thrown by SSviewer.php

**Comment by Friksel** [comment]
Hi @dhensby Thanks for your reply. The templates are within `templates\Shortcodes` inside the theme-folder. I understand the CMS runs without a theme, but I cannot think of a reason why the CMS needs the theme for saving a page, because it doesn't need to parse the shortcode when the page is being saved into database with the shortcode as is. Only when the page is being loaded on the front-website it needs to execute the parser and then the theme is loaded. At least this was working this way for two years...

**Comment by dhensby** [comment]
very good point, and that's probably where the regression is (rendering the shortcode unnecessarily on save). Do you think you can track this down? looking at a backtrace from where the error comes from would be a start.

**Comment by Friksel** [comment]
No, I'm not going to. I already spend way too much time on this trying to debug this. Next to the fact I don't have the time for it I don't know the inner workings of the framework and with version 4 around the corner it would be a waste in spending time on that.
For me it's not worth it to spend all this time to debug the framework which was running fine for years. I don't know what changed in the framework or somewhere else, especially now version 4 is the active developed version, but I'd rather go back to a previous version then.

**Comment by Friksel** [comment]
Hey, I don't like your action. This is not the way it should work on an issuetracker. Why do you close this issue if it's still there? I took the time to report this as a user of Silverstripe Framework, took the time to try to debug problems in the framework myself, which is not even my work in the first place. And now you're closing this issue while it's still there? You really think it's my responsibility to fix a problem in the framework? I hope not. Please reopen the issue. Closing doesn't make any sense.

**Comment by dhensby** [comment]
Hi @Friksel - sorry, I'm not trying to cause a problem.

SilverStripe Framework is a free and open source piece of software, so defining who is responsible for fixing bugs is a bit difficult and ultimately vague. We rely on the community, core committers and sometimes SilverStripe Ltd to fix these issues.

Whilst you may view that fixing the issue yourself is not your work it's also nobody else's (unless they wish to volunteer that time and take it on as their work).

I'm sorry my action of closing the issue is not as you wish, but you expressed a view that the effort and imperative to fix this issue was lacking as 4.x is now available. I interpreted this to mean you were happy to close the issue as it was too minor to invest any effort to resolve.

Whilst this issue is probably present (though I haven't replicated it personally) I think it's quite an edge-case which, as you note, doesn't warrant the effort to debug. Therefore I think it should be closed as "wontfix" because of it's low impact and the fact it's happening in legacy software.

If you'd like to reopen it, I'll leave that to you, but I'm not sure it'll be fixed if it doesn't affect 4.x. If you do reopen it, providing a backtrace would be most helpful (and a required step) in allowing others to begin to resolve this bug.

**Comment by laminbarrow** [comment]
@Friksel I've had this issue as well on an older Silverstripe 3 site. In the end, I **FIXED** it by moving the shortcode templates away from my theme into the `/mysite/templates` directory
